dart/test/server_test.dart

```diff
@@ -251,6 +255,59 @@ void main() {
           }))));
     });
   });
+
+  group('registerService', () {
+    test('can delegate requests between clients', () async {
+      var serviceId = 'ext.test.service';
+      var responseQueue = StreamQueue(responsesController.stream);
+
+      var clientInputController =
+          StreamController<Map<String, Object>>.broadcast();
+      var clientOutputController =
+          StreamController<Map<String, Object>>.broadcast();
+      var client = VmService(clientInputController.stream.map(jsonEncode),
+          (String message) => clientOutputController.add(jsonDecode(message)),
+          disposeHandler: () async {
+        await clientInputController.close();
+        await clientOutputController.close();
+      });
+
+      var clientConnection = VmServerConnection(clientOutputController.stream,
+          clientInputController.sink, serviceRegistry, serviceMock);
+
+      var requestParams = {'foo': 'bar'};
+      var expectedResponse = Response()..json = {'zap': 'zip'};
+      await client.registerService(serviceId, null);
+      // Duplicate registrations should fail.
+      expect(client.registerService(serviceId, null),
```

Comment:
[nit] Each of these calls to expect should really be contained in disjoint tests.

Reply:
I agree that would be ideal - but in this case it is pretty difficult to do in a way that doesn't just cause a ton of duplication.